lib/phase/functional.js

```diff
@@ -237,6 +258,61 @@ function validateWorkflowGraph(doc) {
     return [];
 }
 
+/**
+ * Check there are no duplicate jobs in stages and all jobs listed exist
+ * @method verifyStages
+ * @param  {Object} stages Stages
+ * @param  {Object} jobs   Jobs
+ */
+function verifyStages(stages, jobs) {
+    if (!stages) {
+        return;
+    }
+
+    // Get list of job names in jobs
+    const jobNames = Object.keys(jobs);
+
+    // Get list of job names in stages
+    let stageJobNames = [];
+
+    Object.values(stages).forEach(stage => {
+        stageJobNames = stageJobNames.concat(stage.jobs);
+    });
+
+    // If job name is repeated in stages, throw error
+    const duplicateJobsInStage = stageJobNames.filter((item, index) => stageJobNames.indexOf(item) !== index);
+
+    if (duplicateJobsInStage.length > 0) {
+        throw new YamlParser.YAMLException(`Cannot have duplicate job in multiple stages: ${duplicateJobsInStage}`);
+    }
+
+    // If job name does not exist, throw error
+    const nonexistentJobsInStage = stageJobNames.filter(jobName => !jobNames.includes(jobName));
+
+    if (nonexistentJobsInStage.length > 0) {
+        throw new YamlParser.YAMLException(`Cannot have nonexistent job in stages: ${nonexistentJobsInStage}`);
+    }
+
+    Object.keys(stages).forEach(stageName => {
+        if (!stages[stageName].jobs || stages[stageName].jobs.length === 0) {
+            throw new YamlParser.YAMLException(`Cannot have stage ${stageName} without jobs`);
+        }
```

Comment:
This would have already been captured during structure validation based on the schema
https://github.com/screwdriver-cd/data-schema/blob/master/config/base.js#L44
```suggestion
```